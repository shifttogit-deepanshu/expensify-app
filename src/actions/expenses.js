import database from "../firebase/firebase"
export const addExpense = (expense)=>(
    {   type:"ADD_EXPENSE",
        expense
    }

)

export const startAddExpense = (expenseData = {})=>{
    return (dispatch)=>{
        const {
            description='',
            note='',
            amount=0,
            createdAt=0
        }=expenseData
        const expense = {description,note,amount,createdAt};
    return database.ref('expenses').push(expense).then((ref)=>{
        dispatch(addExpense({
            id:ref.key,
            ...expense
        }))
    })
    }
}

export const removeExpense = ({id=0}={})=>(
    {
        type:"REMOVE_EXPENSE",
        id
    }
)

export const editExpense = (id,updates)=>({
    type:"EDIT_EXPENSE",
    id,
    updates
})

export const setExpense = (expenses)=>({
    type:"SET_EXPENSE",
    expenses
})

export const startSetExpense=()=>{
    return (dispatch)=>{
        return database.ref('expenses').once('value').then((snapshot)=>{
            const expenses = []
            snapshot.forEach((childSnapshot)=>{
                expenses.push({
                   id: childSnapshot.key,
                   ...childSnapshot.val()
                })
            })
            dispatch(setExpense(expenses))
        })
    }
}