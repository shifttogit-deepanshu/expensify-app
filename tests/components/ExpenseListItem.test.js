import React from 'react'
import ExpenseListItem from "../../src/components/ExpenseListItem"
import {shallow} from "enzyme"
import expenses from "../fixtures/expenses"

test('should render expense list item ',()=>{
    const wrapper = shallow(<ExpenseListItem expense={expenses[1]} />)
    expect(wrapper).toMatchSnapshot()
})