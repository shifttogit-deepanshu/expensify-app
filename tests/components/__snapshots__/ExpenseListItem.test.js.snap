// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item  1`] = `
<div>
  <Link
    to="edit/undefined"
  >
    <h3 />
  </Link>
  <p>
    $0.00
     - 
    January 1st, 1970
  </p>
</div>
`;
